src/libcsg/modules/io/xmltopologyreader.cc

```diff
@@ -134,25 +136,58 @@ void XMLTopologyReader::ParseMolecule(Property &p, string molname, int nbeads, i
             } catch (runtime_error) {
                 atq = 0.0;
             }
+            try {
+                resid = it->getAttribute<int>("resid");
+                if (resid<=0)
+                    throw std::runtime_error(
```

Comment:
Thanks for spotting this! It was not intentional and I replaced most of the `std::runtime_error` I added by `std::invalid_argument` as this won't be caught and fits better IMHO. The attached diff `small.txt` should hopefully alleviate this and the coding style issues as well. The attached diff `total.txt` is a diff with respect to the original. Using diffs is quite tedious. Should I create another pull request (I probably should have created one straight away)?

[small.txt](https://github.com/votca/csg/files/940997/small.txt)
[total.txt](https://github.com/votca/csg/files/940999/total.txt)